packages/input/src/ux-input.html

```diff
@@ -22,5 +22,6 @@
 
   <slot name="trailing-icon"></slot>
 
-  <div class="ux-input__border-bottom"></div>
+  <div class="ux-input__border-bottom" if.bind="variant === 'filled' && !disabled"></div>
```

Comment:
Are two `<div/>` necessary? Is there any way we can reduce this to a single `div` and use some thing like show/css class combo?

Reply:
I will explore solutions for this. At first I tried but couldn't find a solution that create nice transition. However now that I think again I think about two ideas:

1. Using `background-image` with several gradient instead of a `background-color`. Not sure how well this property can transition but from memory It could work. Down-side is that the code becomes difficult to read.
2. Using a simple `:before` pseudo-element for the fine border and css with transitions for the larger border.

Will keep you updated soon

Reply:
@bigopon 

I've found a nice fix. I left only one `<div>` in the HTML. The div is used as a container for two borders (fine and large) and they are displayed using `:before` and `:after` pseudo-element respectively.

Fix: https://github.com/platform5/ux/commit/0dade333953a213a1d8804a05d93ad0dba61f829

_If you find this satisfactory, I will apply the same fix to the other components_

Reply:
That looks nice. Can we also try with `show.bind` instead of `if.bind`? I think show/hide would be sufficient, I'm not sure bout this.

Reply:
@bigopon https://github.com/aurelia/ux/pull/232/commits/f38c92a3ad79bcfc8856de1612d0cf579435b089

BTW: can you explain why in this case show.bind would be better than if.bind? Is show.bind better for perf ?

Reply:
Sorry missed this, yes, `show.bind` is a lot more lightweight if the requirement is all about visual, not the dom structure.